include/seqan3/argument_parser/auxiliary.hpp

```diff
@@ -73,38 +73,49 @@ namespace seqan3::detail::adl_only
 template <typename t>
 std::unordered_map<std::string_view, t> enumeration_names(t) = delete;
 
-/*!\brief Functor definition for seqan3::enumeration_names.
- *
- * We need a class template here because we need the original option_t next to s_option_t that may be wrapped
- * std::type_identity to be default constructible. We need option_t to be default constructible because the
- * respective CPO should be callable without a function parameter.
- */
+//!\brief seqan3::detail::customisation_point_object (CPO) definition for seqan3::enumeration_names.
+//!\ingroup alphabet
 template <typename option_t>
-struct enumeration_names_fn
+struct enumeration_names_cpo : public detail::customisation_point_object<enumeration_names_cpo<option_t>, 1>
 {
-    //!\brief `option_t` with cvref removed and possibly wrapped in std::type_identity.
-    using s_option_t = std::conditional_t<std::is_nothrow_default_constructible_v<std::remove_cvref_t<option_t>> &&
-                                          seqan3::is_constexpr_default_constructible_v<std::remove_cvref_t<option_t>>,
-                                          std::remove_cvref_t<option_t>,
-                                          std::type_identity<option_t>>;
+    //!\brief CRTP base class seqan3::detail::customisation_point_object.
+    using base_t = detail::customisation_point_object<enumeration_names_cpo<option_t>, 1>;
+    //!\brief Only this class is allowed to import the constructors from #base_t. (CRTP safety idiom)
```

Comment:
What is #base_t?

Reply:
`#base_t` is a doxygen command to force a link to base_t. Unfortunately, `using base_t::base_t;` isn't shown in the documentation.